### ISSUE TITLE
Typed array constructor is evaluated to verbose array initializer

### ISSUE THREAD

**Issue by Frizi** [facebook/prepack#557]
Evaluation of typed array constructor lead to weird code expansion:

Input:
```javascript
(function() {
  global.result = new Uint8Array(10);
})();
```
output:
```javascript
(function () {
  var _$0 = Uint8Array;
  result = new _$0([0, 0, 0, 0, 0, 0, 0, 0, 0, 0]);
})();
```

While the code is still equivalent in behaviour, the it's not only more verbose but also much slower. I'd actually expect the output to be roughly the same as input.

**Comment by NTillmann** [comment]
The serializer could certainly look for some patterns and emit better code.

**Comment by NTillmann** [comment]
Optimization implemented as part of #636.
